### DISCUSSION TITLE
Remove unmaintaned packages for more name choices for developers

### DISCUSSION THREAD

**Discussion by DanieliusDev** [npm/feedback#732] [Registry] [answered]
Every time im looking for a name for my npm package, i run into a common problem of finding a package which is has its last update 2+ years ago and hasnt been downloaded by much people which shows the package is practically there for nothing. That is why i think NPM should take action on packages that are
1. Not maintaned
2. Not used

For a start i have 2 ways you can check if a package is needed, feel free to comment under this to give more ideas :)

1. Check the tine period of when the last version of a package was released (example: If a package was updated more than a year ago, it should be considered unmaintained)
2. Check the download numbers for a time period (exanple: if a package doesn't have 50+ downloads in the past 3 or 6 months, it should be considered unused)

**Comment by ljharb** [comment]
The choices here are to either pick a different name (there's infinite ones available) or to use a scope (like `@gamertike/whatever`)

**Comment by ichenglin** [comment]
It's not a good idea to wipe out the inactive packages, programs that **installed them previously** as **dependencies** would run into problems.

For example, the **[FileSystem](https://www.npmjs.com/package/fs)** is widely used for managing files, yet the package **name** is currently **not in use**.
If someone gets to claim it, they could easily **spread malicious code** without others **noticing**.

If you really can't think of another name, it's best to **publish it under scope** just like what [liharb suggested](https://github.com/npm/feedback/discussions/732#discussioncomment-3163349).

**Comment by MylesBorins** [comment, marked as answer]
The registry is generally immutable. When things are published we do not remove is unless it contains malicious code. Very rarely we do hand over a disputed name if someone has a valid claim to it, but if the package has existing published versions we generally do not unpublish those versions.

While it is frustrating that many "great" names have been taken over time, we are not exploring solutions to "free up" those names.
